dotCMS/src/main/java/com/dotcms/storage/StorageKey.java

```diff
@@ -0,0 +1,64 @@
+package com.dotcms.storage;
+
+public class StorageKey {
+
+    private final String path;     // todo: rename to key
+    private final String bucket;  // rename to group
+    private final String storage;
+
+    private StorageKey(final Builder builder) {
+
+        this.path       = builder.path;
+        this.bucket     = builder.bucket;
+        this.storage    = builder.storage;
+    }
+
+    public String getPath() {
+        return path;
+    }
+
+    public String getBucket() {
+        return bucket;
+    }
+
+    public String getStorage() {
+        return storage;
+    }
+
+    public static final class Builder {
+
+        private String path;
+        private String bucket;
+        private String storage;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Field storage has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)